lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3637,10 +3636,15 @@ namespace {
           if (unannotatedAPIWarningNeeded &&
               Impl.SwiftContext.LangOpts.hasFeature(
                   Feature::WarnUnannotatedReturnOfCxxFrt)) {
-            Impl.addImportDiagnostic(
-                decl,
-                Diagnostic(diag::no_returns_retained_returns_unretained, decl),
-                decl->getLocation());
+            ASTContext &ctx = Impl.SwiftContext;
+            std::string msg =
+                "It should be annotated with either SWIFT_RETURNS_RETAINED or "
+                "SWIFT_RETURNS_UNRETAINED as it is returning a "
+                "SWIFT_SHARED_REFERENCE";
+            auto message = ctx.AllocateCopy(StringRef(msg));
```

Comment:
I think you can also get rid of `ctx.AllocateCopy` since we are dealing with a static string here.